pkg/api/agentPoolOnlyApi/v20180331/errors.go

```diff
@@ -0,0 +1,45 @@
+package v20180331

```

Comment:
Why does this need a new `errors.go` file? Can you consolidate this where other errors like `ErrorResourceGroupNotMatch` are?

Reply:
I separate the errors out and make them each a variable, so I can catch the exact one when running unit tests. Otherwise, the returned error from unit tests will be a generic error and I will need to match the wording in the error object.

Please let me know if you have a better suggestion for such error handling.

Reply:
Talked with Jiantian and his error case is different. So for now, will keep as it is. Later we can think about error standardization/centralization.